### PR TITLE
Added initial finetuning

### PR DESCRIPTION
**Pull request by cssprad1**
# pytorch-caney - Adding Satvision/MiM/SWIN finetuning

## Description
This PR adds the code base to enable finetuning of SWIN vision transformers. The implementation is broken up into several sub-modules. I tried best to match the style of of PTC is set up. 

<i>Note: this is not using pytorch lighting. I would like to use it at some point and/or give the user the option to run through pt-lightning instead of pure pytorch w/ DDP</i>

## Testing
Needs independent testing

### CODE REVIEW

File: pytorch_caney/pipelines/pretraining/simmim.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
 def parse_args():
     parser = argparse.ArgumentParser(
-        'pytorch-caney impletmentation of SimMiM pre-training script',
+        'pytorch-caney impletmentation of MiM pre-training script',
```
Comment (jordancaraballo) [inline pytorch_caney/pipelines/pretraining/simmim.py:32]: implementation instead of impletmentation

Comment (jli-99) [reply pytorch_caney/pipelines/pretraining/simmim.py:32]: so does the name of the script need to be changed to "mim.py"?

**Review by jli-99 — CHANGES_REQUESTED** [review]

File: pytorch_caney/pipelines/pretraining/simmim.py (hunk 1 above)
Comment (cssprad1) [reply pytorch_caney/pipelines/pretraining/simmim.py]: Fixed spelling in https://github.com/nasa-nccs-hpda/pytorch-caney/pull/22/commits/41b53be4f3b47e52a052d61012e2d881b1d12106

Comment (cssprad1) [reply pytorch_caney/pipelines/pretraining/simmim.py]: Changed name of script in https://github.com/nasa-nccs-hpda/pytorch-caney/pull/22/commits/1632f189ded29934679cbdbd694210a067654e47

**Pull request merged**